src/render.lisp

```diff
@@ -170,6 +170,35 @@ about the specified renderer, and return it."
 (defun get-renderer-output-size (renderer)
   (niy "SDL_GetRendererOutputSize()"))
 
+(defun query-texture (texture &key format access width heigth)
+  (with-foreign-objects ((format-ptr :uint32)
+			 (access-ptr :int)
+			 (width-ptr  :int)
+			 (heigth-ptr :int))
+    (check-rc (sdl-query-texture texture format-ptr access-ptr width-ptr heigth-ptr))
+    (values (mem-ref format-ptr :uint32)
+	    (mem-ref access-ptr :int)
+	    (mem-ref width-ptr  :int)
+	    (mem-ref heigth-ptr :int))))
+
```

Comment:
Using `C-WITH` here instead of `WITH-FOREIGN-OBJECTS` is easier, it looks something like:

```lisp
(c-with ((format uint32-t)
         (access :int)
         (width  :int)
         (height :int))
  (check-rc (sdl-query-texture texture (format &) (access &) (width &) (height &)))
  (values format access width height))
```